src/modules/claim/resolvers/credential.resolver.ts

```diff
@@ -1,69 +1,104 @@
-import jsonwebtoken from 'jsonwebtoken';
 import {
   CredentialResolver,
+  isEIP191Jwt,
+  isVerifiableCredential,
   RoleEIP191JWT,
+  transformClaim,
   VerifiableCredential,
 } from '@energyweb/vc-verification';
-import { ClaimData } from '@energyweb/vc-verification/dist/src/models';
+import { RolePayload } from '@energyweb/vc-verification';
 import { RoleCredentialSubject } from '@energyweb/credential-governance';
-import { IPublicClaim } from '@ew-did-registry/claims';
-import { ClaimService } from '../services';
-import { RegistrationTypes } from '../claim.types';
+import * as jwt from 'jsonwebtoken';
+import { DIDService } from '../../did/did.service';
 
 export class RoleCredentialResolver implements CredentialResolver {
-  constructor(private readonly claimService: ClaimService) {}
+  constructor(private readonly didService: DIDService) {}
+
   async getCredential(
     did: string,
     namespace: string
   ): Promise<
     VerifiableCredential<RoleCredentialSubject> | RoleEIP191JWT | null
   > {
+    const tokens = await this.didService.resolveServiceEndpoints(did);
     return (
-      (await this.getVerifiableCredential(did, namespace)) ||
-      (await this.getEIP191JWT(did, namespace))
+      this.serviceEndpointsToCredentials(tokens).find(
+        (cred) => cred.credentialSubject.role.namespace === namespace
+      ) ||
+      this.serviceEndpointsToEIP191(tokens).find(
+        (token) => token.payload.claimData.claimType === namespace
+      )
```

Comment:
Maybe it's good to have the optional chaining operator here in order to avoid an exception? I think it's best to have each individual method be as safe as possible.
```suggestion
      this.serviceEndpointsToCredentials(tokens).find(
        (cred) => cred?.credentialSubject?.role?.namespace === namespace
      ) ||
      this.serviceEndpointsToEIP191(tokens).find(
        (token) => token?.payload?.claimData?.claimType === namespace
      )
```

Reply:
I agree that credential needs this check, because `isVerifialbleCredential` checks only high-level properties https://github.com/energywebfoundation/ew-credentials/blob/19108ef86410e85aad71bc47d44ddf202ced716f/packages/vc-verification/src/utils/formatters.ts#L108, but for eip191 this is not needed, because whole chain is checked by `isEIP191Jwt` https://github.com/energywebfoundation/ew-credentials/blob/19108ef86410e85aad71bc47d44ddf202ced716f/packages/vc-verification/src/utils/formatters.ts#L97